libcudacxx/include/cuda/discard_memory

```diff
@@ -0,0 +1,165 @@
+ * residents, without regard to the conflicts of laws principles. The United Nations Convention on Contracts for the
+ * International Sale of Goods is specifically disclaimed. You agree to all terms of this Agreement in the English
+ * language. The state or federal courts residing in Santa Clara County, California shall have exclusive jurisdiction
+ * over any dispute or claim arising out of this license. Notwithstanding this, you agree that NVIDIA shall still be
+ * allowed to apply for injunctive remedies or an equivalent type of urgent legal relief in any jurisdiction.
+ *
+ * 12. NO ASSIGNMENT. This license and your rights and obligations thereunder may not be assigned by you by any means or
+ * operation of law without NVIDIA’s permission. Any attempted assignment not approved by NVIDIA in writing shall be
+ * void and of no effect.
+ *
+ * 13. EXPORT. The SOFTWARE is subject to United States export laws and regulations. You agree that you will not ship,
+ * transfer or export the SOFTWARE into any country, or use the SOFTWARE in any manner, prohibited by the United States
+ * Bureau of Industry and Security or economic sanctions regulations administered by the U.S. Department of Treasury’s
+ * Office of Foreign Assets Control (OFAC), or any applicable export laws, restrictions or regulations. These laws
+ * include restrictions on destinations, end users and end use. By accepting this license, you confirm that you are not
+ * a resident or citizen of any country currently embargoed by the U.S. and that you are not otherwise prohibited from
+ * receiving the SOFTWARE.
+ *
+ * 14. GOVERNMENT USE. The SOFTWARE has been developed entirely at private expense and is “commercial items” consisting
+ * of “commercial computer software” and “commercial computer software documentation” provided with RESTRICTED RIGHTS.
+ * Use, duplication or disclosure by the U.S. Government or a U.S. Government subcontractor is subject to the
+ * restrictions in this license pursuant to DFARS 227.7202-3(a) or as set forth in subparagraphs (b)(1) and (2) of the
+ * Commercial Computer Software - Restricted Rights clause at FAR 52.227-19, as applicable. Contractor/manufacturer is
+ * NVIDIA, 2788 San Tomas Expressway, Santa Clara, CA 95051.
+ *
+ * 15. ENTIRE AGREEMENT. This license is the final, complete and exclusive agreement between the parties relating to the
+ * subject matter of this license and supersedes all prior or contemporaneous understandings and agreements relating to
+ * this subject matter, whether oral or written. If any court of competent jurisdiction determines that any provision of
+ * this license is illegal, invalid or unenforceable, the remaining provisions will remain in full force and effect.
+ * This license may only be modified in a writing signed by an authorized representative of each party.
+ *
+ * (v. August 20, 2021)
+ */
```

Comment:
This should use our shortened license similar to other headers

Reply:
Thanks, I switched to the shortened license 👍